### PR TITLE
Refactored CAL library to include encoding

### PR DESCRIPTION
**Pull request by Jchisholm204**
Refactored decoder into encoder class, then refactored encoder into CAL library.  Split up files for readability.

CAL now stores data within the class constructor in  5 CAN_msg_t objects dedicated to the 5 CAN ID's the library can currently encode/decode.

Current compilation warnings are for functions that will not return anything on their default constructor, but the way the library is built this is not possible via their input therefore this can be ignored.

Also merged into the library are Mock ECU and Mock Steering examples that run on Arduinos, simulating the data transfer between the ECU and the Steering Wheel. Currently, the examples only support shifting.

### CODE REVIEW

File: cal/.vscode/launch.json [hunk 1]
```diff
@@ -0,0 +1,44 @@
+// AUTOMATICALLY GENERATED FILE. PLEASE DO NOT MODIFY IT MANUALLY
```
Comment (ethanmpeterson) [inline cal/.vscode/launch.json]: git ignore vscode directory

File: cal/cal/examples/FakeECU/src/main.cpp [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+ * 
+ * @copyright Copyright (c) 2023
+ * 
+ */
+#include <Arduino.h>
+#include <SPI.h>
+#include "mcp2515_can.h"
+#include <cal.hpp>
+
+mcp2515_can can(9);
```
Comment (ethanmpeterson) [inline cal/cal/examples/FakeECU/src/main.cpp]: avoid magic numbers

File: cal/cal/examples/FakeECU/src/main.cpp [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+CAL::CAL cal;
+
+
+void setup() {
+  Serial.begin(9600);
+  while(can.begin(CAN_500KBPS) != CAN_OK){
```
Comment (ethanmpeterson) [inline cal/cal/examples/FakeECU/src/main.cpp]: Should we just standardize on 1MBPS?

**Review by ethanmpeterson — COMMENTED** [review]
- remember to remove checked in can bus library. maybe add a global git ignore for pio build and vscode directory junk
-

File: cal/.vscode/launch.json (hunk 1 above)
Comment (Jchisholm204) [reply cal/.vscode/launch.json]: I thought we were leaving it because all it is is the extension prompt

**Comment by Jchisholm204** [comment]
OK, we are all good

**Review by llim1 — APPROVED** [review]

**Pull request merged**